app/models/job_invocation_composer.rb

```diff
@@ -79,7 +79,7 @@ def triggering
       return {} unless ui_params.key?(:triggering)
       trig = ui_params[:triggering]
       keys = (1..5).map { |i| "end_time(#{i}i)" }
-      return trig unless trig.key?(:end_time) && trig[:end_time].keys == keys
+      return trig unless trig.key?(:end_time) && trig[:end_time].keys.sort == keys
       trig.merge(:end_time => Time.local(*trig[:end_time].values_at(*keys)))
```

Comment:
I know I originally suggested the sorting as an easy way out, but now I'm having second thougths about it. How about something like this
```ruby
values = trig.fetch(:end_time, {}).values_at(*keys)
return trig if values.any?(&:nil?)
trig.merge(:end_time => Time.local(*values))
```

Reply:
Yeah it makes more sense, thanks